ipykernel/kernelbase.py

```diff
@@ -1131,9 +1136,62 @@ def _input_request(self, prompt, ident, parent, password=False):
             raise EOFError
         return value
 
-    def _at_shutdown(self):
+    async def _progressively_terminate_all_children(self):
+        if sys.platform == "win32":
```

Comment:
I think this method should work on both Windows and Posix: https://stackoverflow.com/a/25134985

Reply:
I'll give it a try. I know that the "loop and kill children" was switched back to `killpg` at some point in jupyter_client as the first one had issues.

Reply:
(that was in jupyter/jupyter_client#743, i'll try to get the same logic from there)